### ISSUE TITLE
id: Illegal option on macOS

### ISSUE THREAD

**Issue by sgade** [rafaelrinaldi/pure#260]
# Bug description

Hello,

I noticed an error message after updating that is displayed every time the prompt is rendered:

```shell
id: illegal option -- -
usage: id [user]
       id -A
       id -F [user]
       id -G [-n] [user]
       id -M
       id -P [user]
       id -g [-nr] [user]
       id -p [user]
       id -u [-nr] [user]
```

The origin seems to be commit a80cb551b233f8ce3299f0c6a1961a24c53ed1b7 which uses flags that are not supported on macOS.
`man id` lists the following:

```
NAME
     id -- return user identity

SYNOPSIS
     id [user]
     id -A
     id -F [user]
     id -G [-n] [user]
     id -M
     id -P [user]
     id -g [-nr] [user]
     id -p [user]
     id -u [-nr] [user]
```

# Proposed solution

At the bottom, it even lists the command that should be used in this context. I tested it, it returns the current user's username.
```
The historic whoami(1) command is equivalent to ``id -un''.
```

Would be pleased if you could fix the command invocation quickly, as it is very annoying in every single output and even appears after CTRL+L.
If you need any further information or testing, please let me know! Happy to help.

Appreciate your time.

## Env

**Pure**

    $ echo $pure_version
    3.3.4

**Fish**

    fish --version
    fish, version 3.1.2

**OS/distribution**

    lsb_release -a
macOS 11.1 (current as of time of writing)

**Comment by edouard-lopez** [comment]
@sgade thanks for the detailed issue. Please update to 3.3.5 as it's been fixed

    fisher install rafaelrinaldi/pure@v3.3.5

**Comment by jorgebucaran** [comment]
There's no need to specify a tag if this is already in the main branch.

**Comment by sgade** [comment]
@edouard-lopez Thanks for your reply.
I updated to `3.4.1` and am still getting the same error message. Could you please check?

**Comment by jorgebucaran** [comment]
I think @edouard-lopez meant to downgrade to @3.3.5 until a new release with the fix is out.

**Comment by edouard-lopez** [comment]
@sgade stick to 3.3.5, I'm working on #259 to fix thing on macOS

**Comment by sgade** [comment]
Alright, will do.

**Comment by sgade** [comment]
The fix in 3.4.1 works. 👍  Thank you, @edouard-lopez.
